From: Jan Engelhardt <jengelh@inai.de>
Date: 2018-04-11 20:00:58.584068848 +0200
References: https://sourceware.org/bugzilla/show_bug.cgi?id=23048

libsiconv can/will use libc iconv if it knows about a particular
character set (or pair thereof), and will fall back to its own
implementation and tables otherwise. glibc fails to offer the
"cp10000" identifier, but it does have the desired table under a
different name.

Change the name in .c so we do not need to install the cp10000 siconv
table.

---
 mkisofs/mkisofs.8 |   13 ++++++-------
 mkisofs/mkisofs.c |    4 ++--
 2 files changed, 8 insertions(+), 9 deletions(-)

Index: schily-2022-10-16/mkisofs/mkisofs.8
===================================================================
--- schily-2022-10-16.orig/mkisofs/mkisofs.8
+++ schily-2022-10-16/mkisofs/mkisofs.8
@@ -2500,8 +2500,8 @@ Input charset that defines the character
 used with the
 .I \-mac\-name
 option.
-The default charset is cp10000 (Mac Roman)
-.I cp10000
+The default charset is
+.I macintosh
 (Mac Roman)
 See
 .B "CHARACTER SETS"
@@ -2711,7 +2711,7 @@ any of the various Apple/Unix file forma
 option. See the
 .B HFS MACINTOSH FILE NAMES
 for more information. Defaults to
-.I cp10000
+.I macintosh
 (Mac Roman).
 .TP
 .B \-output\-hfs\-charset
@@ -3227,13 +3227,12 @@ name will be used in the TRANS.TBL file,
 .PP
 The character set used to convert any HFS file name to a Joliet/Rock Ridge
 file name defaults to
-.I cp10000
+.I macintosh
 (Mac Roman).
 The character set used can be specified using the
 .I \-input\-hfs\-charset
-option. Other built in HFS character sets are: cp10006 (MacGreek),
-cp10007 (MacCyrillic), cp10029 (MacLatin2), cp10079 (MacIcelandic) and
-cp10081 (MacTurkish).
+option. Other built in HFS character sets are:
+MAC-CYRILLIC (MacCyrillic), MAC-CENTRALEUROPE (MacLatin2), MAC-IS (MacIcelandic).
 .PP
 Note: the character codes used by HFS file names taken from the various
 Apple/Unix formats will not be converted as they are assumed to be in the
Index: schily-2022-10-16/mkisofs/mkisofs.c
===================================================================
--- schily-2022-10-16.orig/mkisofs/mkisofs.c
+++ schily-2022-10-16/mkisofs/mkisofs.c
@@ -2888,7 +2888,7 @@ setcharset:
 
 #ifdef APPLE_HYB
 	if (hfs_icharset == NULL || strcmp(hfs_icharset, "mac-roman") == 0) {
-		hfs_icharset = "cp10000";
+		hfs_icharset = "macintosh";
 	}
 	hfs_inls = sic_open(hfs_icharset);
 
@@ -2896,7 +2896,7 @@ setcharset:
 		hfs_ocharset = hfs_inls ? hfs_inls->sic_name : NULL;
 	}
 	if (hfs_ocharset == NULL || strcmp(hfs_ocharset, "mac-roman") == 0) {
-		hfs_ocharset = "cp10000";
+		hfs_ocharset = "macintosh";
 	}
 	hfs_onls = sic_open(hfs_ocharset);
 
